client/src/panels/decompositionPanel.js

```diff
@@ -104,6 +105,32 @@ class DecompositionPanel extends React.Component {
     this.updateDataModel('' + result.maxPersistenceLevel);
   }
 
+  componentDidMount() {
+    if (this.state.decompositionMode == 'Morse-Smale') {
+      let k = 15; // num nearest neighbors to consider when generating M-S complex for a dataset
+      let datasetId = this.props.dataset.datasetId;
+      let category = this.state.decompositionCategory;
+      let field = this.state.decompositionField;
+      this.client.fetchMorseSmalePersistence(datasetId, category, field, k)
+          .then(function(result) {
+            this.setState({
+              minPersistence: result.minPersistenceLevel,
+              maxPersistence: result.maxPersistenceLevel,
+              complexSizes: result.complexSizes,
+              sliderPersistence: result.maxPersistenceLevel,
+              persistenceLevel: ('' + result.maxPersistenceLevel),
+            });
+            this.updateDataModel('' + result.maxPersistenceLevel);
+          }.bind(this));
+    } else {
+      this.setState({
```

Comment:
could start by setting state to these default values, then ask if decompositionMode is M-S and call fetchDecomposition if it is.